bin/VMSlogger

```diff
@@ -120,57 +179,95 @@ class Collector:
             logger.info(f"Closing HDF5 {self.h5file.file.filename}")
             self.h5file.close()
 
-    async def collect_data(self):
-        async with Domain() as domain:
-            remote = Remote(domain, "MTVMS", index=self.index + 1)
-            remote.tel_data.callback = lambda data: self.cache.newChunk(data, 0.001)
-            if logger.getEffectiveLevel() == logging.DEBUG:
-                while not (self.cache.filled):
-                    await asyncio.sleep(0.5)
-                    logger.debug(
-                        f"Waiting {devices[self.index]}..{100 * len(self.cache)/self.size:.02f}% {len(self.cache)} of {self.size}"
-                    )
-                    self._save_hdf5()
+    async def sample_file(self):
+        if self.daemonized or logger.getEffectiveLevel() == logging.DEBUG:
+            saved_len = 0
+            while True:
+                current_len = saved_len + len(self.cache)
+                logger.debug(
+                    f"Waiting {devices[self.index]}..{100 * (current_len)/self.size:.02f}% {current_len} of {self.size}"
+                )
+                if self.h5file is None:
+                    if current_len >= self.size:
+                        break
+                else:
+                    if self._save_hdf5():
+                        saved_len += self.chunk_size
```

Comment:
Please consider to refactor the code to decrease the indentation if possible.

Reply:
Thanks, I considered splitting that before, finally made the split - see last comment.